pkg/course/course.go

```diff
@@ -261,14 +263,25 @@ func OpenCourseV2(fileName string, schema []byte) (*FileV2, error) {
 	}
 	if courseFile.SchemaVersion != "v2" {
 		klog.V(2).Infof("did not detect v2 course file - trying conversion from v1")
-		fileV2, errConvert := ConvertV1toV2(fileName)
+		fileV2, errConvert := convertV1toV2(fileName)
 		if errConvert != nil {
-			return nil, fmt.Errorf("could not unmarshal file from v1 or v2 schema")
+			return nil, fmt.Errorf("could not unmarshal file from v1 or v2 schema:\n\t%s", errConvert.Error())
 		}
 		color.Yellow("WARNING: this course file was automatically converted from v1 to v2 at runtime - to convert the file permanently, run \"reckoner convert -i %s\"", fileName)
 		courseFile = fileV2
 	}
 
+	// Marshal back here just so we can populate the env vars without any yaml comments present
+	data, _ = yaml.Marshal(courseFile)
+	data, err = parseEnv(data)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse env variables: %v", err)
+	}
+	err = yaml.Unmarshal(data, courseFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal file after parsing env vars: %s", err.Error())
+	}
+
```

Comment:
The reason we are doing this unmarshal -> marshal -> unmarshal again is because if we parsed environment variables before the first unmarshal, it would also error out on any YAML comments that mention an environment variable. When we do the first unmarshal, all comments are stripped so we can marshal back to bytes, populate all the env vars and then unmarshal into our struct again.

Reply:
That's actually really cool. That was fairly annoying in the python version at one point.